corehq/apps/users/models.py

```diff
@@ -2625,6 +2625,7 @@ class SQLInvitation(SyncSQLToCouchMixin, models.Model):
     program = models.CharField(max_length=126, null=True)   # couch id of a Program
     supply_point = models.CharField(max_length=126, null=True)  # couch id of a Location
     couch_id = models.CharField(max_length=126, null=True, db_index=True)
+    uuid = models.UUIDField(unique=True, db_index=True, default=uuid4)
```

Comment:
can we use `invitation_id` instead of uuid to keep in line with other models like SQLLocation?
this way we can also just keep the variable name `invitation_id` in other places that are getting updated.

Reply:
I find it confusing that there's both a `SQLLocation.id` and a `SQLLocation.location_id`. Is that really a best practice that we use elsewhere?

Reply:
I think we follow different things but introducing a new one might not be a good thing.
in some other instances,`ModuleBase` has `unique_id`, `CommCareCaseSQL` has `case_id`.

Reply:
Okay, I'm inclined to go with `unique_id` then. Agreed introducing a new convention isn't good.

Reply:
> I find it confusing that there's both a SQLLocation.id and a SQLLocation.location_id. Is that really a best practice that we use elsewhere?

I definitely wouldn't call it a best practice.  And yes, it is confusing.  I only ever refer to `location.id` as `location.pk` in order to help with the ambiguity.  My rule of thumb is that you should pretty much only ever use `location_id` unless you're doing something low-level and db-related that needs access to the primary key (for instance looking across foreign key relationships without needing to do a join).

My recommendation would be to set `primary_key=True` on `uuid`, which will allow Django to drop `id` entirely.  Any place which would reference it would then reference `uuid` instead.  Do you know whether any other models have a foreign key reference to this one?  That would complicate the migration (that's why I never did it for locations).

Reply:
No strong opinions on `unique_id` vs `uuid` vs `invitation_id`, though I do agree that following convention would be best.  You may be in a good position to _establish_ such a convention, actually.

Looking at the lines matching `id.=.*index` in `models.py` files, all results are <noun>_id.  There don't seem to be many models with a field specified as `primary_key=True`, so arguably that's a different concern.

Actually, after writing that out, I do have opinions - I like `uuid` the most, as I'd expect `id` to be a sequential ID.  `unique_id` is also reasonable.  `<noun>_id` feels like it should be used for related models, not as a primary key field (typing `location.location_id` makes my eye twitch).

Reply:
> My recommendation would be to set `primary_key=True` on `uuid`

If you do this, please change the name to `id`